src/libraries/System.Private.CoreLib/src/System/Reflection/InvokerEmitUtil.cs

```diff
@@ -0,0 +1,331 @@
+        /// Push each argument.
+        /// </summary>
+        private static void PushArguments(
+            ILGenerator il,
+            ParameterInfo[] parameters,
+            out NullableRefInfo[]? byRefNullables,
+            out int byRefNullablesCount
+            )
+        {
+            byRefNullables = null;
+            byRefNullablesCount = 0;
+
+            LocalBuilder? local_pArg = il.DeclareLocal(typeof(IntPtr*));
+            il.Emit(OpCodes.Ldarg_2);
+            il.Emit(OpCodes.Stloc_S, local_pArg);
+
+            for (int i = 0; i < parameters.Length; i++)
+            {
+                RuntimeType parameterType = (RuntimeType)parameters[i].ParameterType;
+
+                // Get the argument as a ref
+                il.Emit(OpCodes.Ldloc_S, local_pArg);
+                il.Emit(OpCodes.Ldind_Ref);
+
+                if (parameterType.IsByRef)
+                {
+                    RuntimeType elementType = (RuntimeType)parameterType.GetElementType();
+
+                    if (elementType.IsNullableOfT)
+                    {
+                        byRefNullables ??= new NullableRefInfo[MaxArgumentCount - i];
+
+                        LocalBuilder tmp = il.DeclareLocal(typeof(object).MakeByRefType());
+                        il.Emit(OpCodes.Stloc_S, tmp);
+                        il.Emit(OpCodes.Ldloc_S, tmp);
+
+                        // Get the raw pointer.
+                        il.Emit(OpCodes.Ldobj, typeof(object));
+                        il.Emit(OpCodes.Unbox, elementType);
+
+                        // Copy the pointer to the temp variable and load as a ref.
+                        LocalBuilder byRefPtr = il.DeclareLocal(elementType.MakePointerType());
+                        il.Emit(OpCodes.Stloc_S, byRefPtr);
+                        il.Emit(OpCodes.Ldloca_S, byRefPtr);
+                        il.Emit(OpCodes.Ldind_Ref);
+
+                        byRefNullables[byRefNullablesCount++] = new NullableRefInfo { ParameterIndex = i, LocalIndex = byRefPtr.LocalIndex };
+                    }
+                    else if (elementType.IsPointer)
+                    {
+                        LocalBuilder tmp = il.DeclareLocal(typeof(IntPtr).MakeByRefType());
+                        il.Emit(OpCodes.Stloc_S, tmp);
+                        il.Emit(OpCodes.Ldloc_S, tmp);
+                        il.Emit(OpCodes.Ldobj, typeof(IntPtr).MakeByRefType());
+                    }
+                    else
+                    {
+                        LocalBuilder tmp = il.DeclareLocal(parameterType);
+                        il.Emit(OpCodes.Stloc_S, tmp);
+                        il.Emit(OpCodes.Ldloca_S, tmp);
+                        il.Emit(OpCodes.Ldind_Ref); //keep this or remove and use ldloca?
+                    }
+                }
+                else if (parameterType.IsNullableOfT)
+                {
+                    // Nullable<T> is the only incoming value type that is boxed.
```

Comment:
This code will be changed. I'm going to do a forced boxing of an actual `Nullable<T>` (not `null` or `T`) on the managed side which will remove this special case.